src/main/java/rx/internal/schedulers/NewThreadWorker.java

```diff
@@ -112,7 +124,7 @@ static void purgeExecutors() {
      * @return true if the policy was successfully enabled 
      */
     public static boolean tryEnableCancelPolicy(ScheduledExecutorService exec) {
-        if (!PURGE_FORCE) {
+        if (!PURGE_FORCE && (!IS_ANDROID || IS_ANDROID_AND_SUPPORTS_SET_REMOVE_ON_CANCEL_POLICY)) {
```

Comment:
Also, I used `int` for Android API Version. And as far as I saw via debugging this method called in many cases, so this PR should improve performance of RxJava in many apps (backends and clients).